app/de/frosner/broccoli/services/NomadService.scala

```diff
@@ -117,43 +119,48 @@ class NomadService @Inject()(configuration: Configuration, consulService: Consul
         jobStatuses = idsAndStatusesWithPeriodic.toMap
         filteredIdsAndStatuses.map { case (key, value) => key }.foreach(requestServices)
       }
-      case Failure(throwable) => {
+      case Failure(throwable: ConnectException) => {
         Logger.error(
           s"Failed to request statuses for ${instanceIds.mkString(", ")} from ${jobsRequest.uri}: $throwable")
         setNomadNotReachable()
       }
+      case Failure(throwable) => {
+        Logger.error(
+          s"Failed to request statuses for ${instanceIds.mkString(", ")} from ${jobsRequest.uri}: $throwable")
+      }
     }
   }
 
   // TODO should this go back to InstanceService or how? I mean I need to check the existing instances in order to know whether
   // TODO something is stopped (i.e. not present in Nomad). Or we use a cache that let's the stuff expire if you don't get an answer from Nomad
   // https://www.playframework.com/documentation/2.5.x/ScalaCache
 
-  def requestServices(id: String): Unit = {
+  def requestServices(id: String): Future[Seq[String]] = {
     val queryUrl = nomadBaseUrl + s"/v1/job/$id"
     val jobRequest = ws.url(queryUrl)
     val jobResponse = jobRequest.get().map { response =>
       if (response.status == 200) {
-        response.json.as[JsObject]
+        response.json.as[Job]
       } else {
         throw new Exception(s"Received ${response.statusText} (${response.status})")
       }
     }
-    val eventuallyJobServiceIds = jobResponse.map { jsObject =>
-      val services = (jsObject \\ "Services").flatMap(_.as[JsArray].value.map(_.as[JsObject]))
-      services.flatMap { serviceJsObject =>
-        serviceJsObject.value.get("Name").map(_.as[JsString].value)
-      }
+    val eventuallyJobServiceIds = jobResponse.map { job =>
+      val taskGroups = job.taskGroups.getOrElse(Seq.empty)
+      val tasks = taskGroups.flatMap(_.tasks.getOrElse(Seq.empty))
+      val services = tasks.flatMap(_.services.getOrElse(Seq.empty))
+      services.map(_.name)
     }
-    eventuallyJobServiceIds.onComplete {
-      case Success(jobServiceIds) =>
-        Logger.debug(s"${jobRequest.uri} => ${jobServiceIds.mkString(", ")}")
-        consulService.requestServiceStatus(id, jobServiceIds)
-        setNomadReachable()
-      case Failure(throwable) =>
+    val eventuallyRequestedServices = eventuallyJobServiceIds.map { jobServiceIds =>
+      Logger.debug(s"${jobRequest.uri} => ${jobServiceIds.mkString(", ")}")
+      consulService.requestServiceStatus(id, jobServiceIds)
+      jobServiceIds
+    }
+    eventuallyRequestedServices.onFailure {
+      case throwable =>
```

Comment:
Ah, sorry, didn't know `onFailure` used a partial function.  Sorry for the noise